pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamMessageDecoder.java

```diff
@@ -33,14 +34,17 @@
 @InterfaceStability.Stable
 public interface StreamMessageDecoder<T> {
 
-  static final String RECORD_EXTRACTOR_CONFIG_KEY = "recordExtractorClass";
+  String RECORD_EXTRACTOR_CONFIG_KEY = "recordExtractorClass";
 
   /**
-   * Initialize the decoder with decoder properties map, the stream topic name and stream schema
-   * @param props
+   * Initialize the decoder
+   * @param props decoder properties extracted from the {@link StreamConfig}
+   * @param indexingSchema the Pinot schema
+   * @param topicName topic name of the stream
+   * @param fields the fields to be extracted from the record
```

Comment:
it is not clear whether these are the fields that we should pay attention to in the  _source_ (incoming data) or these are the fields that we _need_ in the schema. I suppose your next step is to remove the `schema` argument, so then the doc becomes more important.

Reply:
improved the javadoc. Renamed it to sourceFields. Added TODO for removing Schema and a note to not use it